src/scenic/syntax/veneer.py

```diff
@@ -111,1000 +132,1341 @@
+            'parentOrientation': pos.orientation
+        }
+        new = DelayedArgument({axis, "contactTolerance"}, val)
+    elif isA(pos, OrientedPoint):        # TODO too strict?
+        prop['parentOrientation'] = 3
+        val = lambda self: {
+            'position': pos.relativize(makeOffset(self, (0,0,0), 0, dx, dy, dz)),
+            'parentOrientation': pos.orientation
+        }
+        new = DelayedArgument({axis}, val)
+    else:
+        pos = toVector(pos, f'specifier "{syntax} X" with X not a vector')
+        val = lambda self: {'position': pos.offsetRotated(self.orientation, makeOffset(self, (0,0,0), 0, dx, dy, dz))}
+        new = DelayedArgument({axis, 'orientation'}, val)
+    return Specifier(syntax, prop, new)
 
 def Following(field, dist, fromPt=None):
-	"""The :specifier:`following {F} from {X} for {D}` specifier.
+    """The :specifier:`following {F} from {X} for {D}` specifier.
+
+    Specifies :prop:`position` and :prop:`parentOrientation`, with no dependencies.
+
+    Allowed forms::
+
+        following <field> [from <vector>] for <number>
+
+    If the :grammar:`from <vector>` is omitted, the position of ego is used.
+    """
+    if fromPt is None:
+        fromPt = ego()
+    if not isA(field, VectorField):
+        raise TypeError('"following F" specifier with F not a vector field')
+    fromPt = toVector(fromPt, '"following F from X for D" with X not a vector')
+    dist = toScalar(dist, '"following F for D" with D not a number')
+    pos = field.followFrom(fromPt, dist)
+    orientation = field[pos]
+    return Specifier("Following", {'position': 1, 'parentOrientation': 3},
+                     {'position': pos, 'parentOrientation': orientation})
+
+def Facing(heading):
+    """The :specifier:`facing {X}` polymorphic specifier.
+
+    Specifies :prop:`yaw`, :prop:`pitch`, and :prop:`roll`, depending on :prop:`parentOrientation`,
+    and depending on the form::
+
+        facing <number>     # no dependencies;
+        facing <field>      # depends on 'position'
+    """
+    if isA(heading, VectorField):
+        def helper(context):
+            headingAtPos = heading[context.position]
+            if alwaysGlobalOrientation(context.parentOrientation):
+                orientation = headingAtPos  # simplify expr tree in common case
+            else:
+                orientation = context.parentOrientation.inverse * headingAtPos
+            return {'yaw': orientation.yaw, 'pitch': orientation.pitch, 'roll': orientation.roll}
+
+        return Specifier("Facing", {'yaw': 1, 'pitch': 1, 'roll': 1}, DelayedArgument({'position', 'parentOrientation'}, helper))
+    else:
+        orientation = toOrientation(heading, "facing x with x not a heading or orientation")
+        orientationDeps = requiredProperties(orientation)
+        def helper(context):
+            nonlocal orientation
```

Comment:
This is confusing -- it makes it look like it's important that the helper is mutating the `orientation` variable, but we don't actually do anything with that variable later. I'd just rename the local variable of `helper` to something else.